packages/app/src/features/notifications/components/PresentationNotificationScreen.tsx

```diff
@@ -21,79 +36,145 @@ export function PresentationNotificationScreen({
+                        <YStack px="$3" pb="$3" gap="$2">
+                          <Paragraph variant="sub">The following information will be presented:</Paragraph>
+                          <YStack flexDirection="row" flexWrap="wrap">
+                            {selectedCredential.requestedAttributes.map((a) => (
+                              <Paragraph flexBasis="50%" key={a} variant="annotation" secondary>
+                                • {sanitizeString(a)}
+                              </Paragraph>
+                            ))}
+                          </YStack>
+                        </YStack>
+                      ) : (
+                        <Paragraph px="$3" pb="$3" variant="sub" color="$danger-500">
+                          This credential is not present in your wallet.
+                        </Paragraph>
+                      )}
                     </YStack>
-                  ) : (
-                    <Paragraph px="$3" pb="$3" variant="sub" color="$danger-500">
-                      This credential is not present in your wallet.
-                    </Paragraph>
-                  )}
-                </YStack>
-              </YStack>
-            ))}
+                  </YStack>
+                )
+              })}
+            </YStack>
           </YStack>
+          {submission.areAllSatisfied ? (
+            <DualResponseButtons onAccept={onAccept} onDecline={onDecline} isAccepting={isAccepting} />
+          ) : (
+            <YStack gap="$4">
+              <Paragraph variant="sub" ta="center">
+                You don't have the required credentials to satisfy this request.
+              </Paragraph>
+              <Button.Solid onPress={onDecline}>Close</Button.Solid>
+            </YStack>
+          )}
         </YStack>
-        {submission.areAllSatisfied ? (
-          <DualResponseButtons onAccept={onAccept} onDecline={onDecline} isAccepting={isAccepting} />
-        ) : (
-          <YStack gap="$4">
-            <Paragraph variant="sub" ta="center">
-              You don't have the required credentials to satisfy this request.
-            </Paragraph>
-            <Button.Solid onPress={onDecline}>Close</Button.Solid>
-          </YStack>
-        )}
-      </YStack>
-    </ScrollView>
+      </ScrollView>
+      <Sheet ref={ref} snapPoints={['40%']} onClose={() => setChangeSubmissionCredentialIndex(-1)}>
+        <BottomSheetScrollView>
+          <Stack bg="$white" pb="$4">
+            {currentSubmissionEntry?.credentials.map((c, credentialIndex) => (
+              <CredentialRowCard
+                onPress={() => {
+                  onSelectCredentialForInputDescriptor(currentSubmissionEntry.inputDescriptorId, credentialIndex)
+                  setChangeSubmissionCredentialIndex(-1)
+                }}
+                // The index is stable enough here
+                // biome-ignore lint/suspicious/noArrayIndexKey: <explanation>
+                key={credentialIndex}
```

Comment:
Does the credential not have an id we can use? I learned my lesson a few weeks back when debugging an index used as a key for way too long.

Reply:
I think we can add it. But we don't have it right now in this model. But yeah no reason to not use the id